### PR TITLE
[report] Honour -o/-e/-n from cmdline over preset

### PR DESCRIPTION
**Pull request by pmoravec**
When preset disables a plugin that user wants to manually enable, current merging of options ends up with -e PLUG -n PLUG which makes the plugin still disabled.

apply_options_from_cmdline method must remove the preset options for any plugin that interfere with cmdline options for the plugin.

Resolves: #3195

---
Please place an 'X' inside each '[]' to confirm you adhere to our [Contributor Guidelines](https://github.com/sosreport/sos/wiki/Contribution-Guidelines)

- [X] Is the commit message split over multiple lines and hard-wrapped at 72 characters?
- [X] Is the subject and message clear and concise?
- [X] Does the subject start with **[plugin_name]** if submitting a plugin patch or a **[section_name]** if part of the core sosreport code?
- [X] Does the commit contain a **Signed-off-by: First Lastname <email@example.com>**?
- [X] Are any related Issues or existing PRs [properly referenced](https://docs.github.com/en/issues/tracking-your-work-with-issues/creating-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword) via a Closes (Issue) or Resolved (PR) line?

### CODE REVIEW

**Comment by pmoravec** [comment]
Particular reproducer: on RHOCP preset (which disables `cgroups` plugin), try to enable `cgroups` plugin.

**Review by TurboTurtle — COMMENTED** [review]
Ack for the goal of this change. Just one comment below.

File: sos/component.py [hunk 1]
```diff
@@ -218,9 +218,21 @@ def apply_options_from_cmdline(self, opts):
...
         codict = cmdopts.dict(preset_filter=False)
         for opt, val in codict.items():
-            if opt not in cmdopts.arg_defaults.keys():
+            if opt not in cmdopts.arg_defaults.keys() or val in [None, [], '']:
                 continue
-            if val not in [None, [], ''] and val != opts.arg_defaults[opt]:
+            # A plugin that is [enabled|disabled|only] in cmdopts must
+            # overwrite these three options of itself in opts - reset it first
+            if opt in ["enable_plugins", "skip_plugins", "only_plugins"]:
+                for oopt in ["enable_plugins", "skip_plugins", "only_plugins"]:
```
[the remark this answers is not in the document]
Comment (pmoravec) [reply sos/component.py:226]: We need to evaluate *pairs* of options+values, if present. Like "`-e logs` in cmdopts AND `-n logs` in preset". So for each `cmdopts` that alters plugin enablement, we must check if it is not in a conflict of a *different* option for the same plugin.

[the remark this answers is not in the document]
Comment (TurboTurtle) [reply sos/component.py:226]: Ah, ok - that tracks. Thanks for clearing that up for me.

**Review by TurboTurtle — APPROVED** [review]

**Pull request merged**